BUG: Fix numpy boolean subtraction error in Series.diff

Unprocessable opened the pull request:
This fixes #27755, for more than three years now NumPy has not allowed the subtraction of boolean series. 

TypeError                                 Traceback (most recent call last)
<ipython-input-46-3da3b949c6bd> in <module>
      1 data = pd.Series([0,-1,-2,-3,-4,-3,-2,-1,0,-1,-1,0,-1,-2,-3,-2,0])
      2 filtered = data.between(-2,0, inclusive = True)
----> 3 filtered.diff()
      4 print(filtered)

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\series.py in diff(self, periods)
   2191         dtype: float64
   2192         """
-> 2193         result = algorithms.diff(com.values_from_object(self), periods)
   2194         return self._constructor(result, index=self.index).__finalize__(self)
   2195 

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\algorithms.py in diff(arr, n, axis)
   1817             out_arr[res_indexer] = result
   1818         else:
-> 1819             out_arr[res_indexer] = arr[res_indexer] - arr[lag_indexer]
   1820 
   1821     if is_timedelta:

TypeError: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.

- [x] closes #17294
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

Unprocessable committed:
Fix numpy boolean subtraction error in Series.diff

Numpy no longer allows subtraction of boolean values. Not sure if Numpy version checking should be done before this code...
Note: I haven't actually run this code, feel free to check it, but should be correct.

---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-46-3da3b949c6bd> in <module>
      1 data = pd.Series([0,-1,-2,-3,-4,-3,-2,-1,0,-1,-1,0,-1,-2,-3,-2,0])
      2 filtered = data.between(-2,0, inclusive = True)
----> 3 filtered.diff()
      4 print(filtered)

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\series.py in diff(self, periods)
   2191         dtype: float64
   2192         """
-> 2193         result = algorithms.diff(com.values_from_object(self), periods)
   2194         return self._constructor(result, index=self.index).__finalize__(self)
   2195 

~\AppData\Local\Continuum\anaconda3\lib\site-packages\pandas\core\algorithms.py in diff(arr, n, axis)
   1817             out_arr[res_indexer] = result
   1818         else:
-> 1819             out_arr[res_indexer] = arr[res_indexer] - arr[lag_indexer]
   1820 
   1821     if is_timedelta:

TypeError: numpy boolean subtract, the `-` operator, is deprecated, use the bitwise_xor, the `^` operator, or the logical_xor function instead.

jreback reviewed (changes requested):
tests are always the first thing we write, pls add them

Unprocessable committed:
Added is_bool outside if function

cbrnr committed:
Slightly rephrase SPSS doc (#27754)

Commit:
DOC: update compiling instructions and link. (#27717)

Unprocessable committed:
Added test_diff function

Unprocessable committed:
Update test_diff.py

pep8speaks commented:
Hello @Unprocessable! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:



* In the file [`pandas/tests/series/test_diff.py`](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py):

> [Line 7:1](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L7): [E302](https://duckduckgo.com/?q=pep8%20E302) expected 2 blank lines, found 1
> [Line 8:21](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:24](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:27](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:30](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:33](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:36](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:39](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:42](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:44](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:47](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:50](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:52](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:55](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:58](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:61](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 8:64](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L8): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 9:31](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L9): [E231](https://duckduckgo.com/?q=pep8%20E231) missing whitespace after ','
> [Line 9:44](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L9): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 9:46](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L9): [E251](https://duckduckgo.com/?q=pep8%20E251) unexpected spaces around keyword / parameter equals
> [Line 11:89](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L11): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (144 > 88 characters)
> [Line 14:89](https://github.com/pandas-dev/pandas/blob/83907a2f1cb5ec2a4e04f22196af7b5fe41e6f22/pandas/tests/series/test_diff.py#L14): [E501](https://duckduckgo.com/?q=pep8%20E501) line too long (121 > 88 characters)

Unprocessable committed:
Changing style to match PEP

Unprocessable committed:
Update test_diff.py

TomAugspurger commented on pandas/tests/series/test_diff.py:
```diff
@@ -0,0 +1,24 @@
+
+
+def test_diff():
+    data = Series(
```
Does it need to be this long?

TomAugspurger commented on pandas/tests/series/test_diff.py:
```diff
@@ -0,0 +1,24 @@
+        [0, -1, -2, -3, -4, -3, -2, -1, 0, -1, -1, 0, -1, -2, -3, -2, 0]
+    )
+    filtered = data.between(-2, 0, inclusive=True)
+    diff_boolean = filtered.diff()
```
Can you remove this and just make the assert on the the result of `data.between`? I'm not sure why `diff` is needed.

TomAugspurger replied:
And use `tm.assert_series_equal` instead of `assert`

Unprocessable replied:
Alright, first comment, release notes, do you have a link for how to write that?
Second comment, agreed, I'll shorten it.
Third comment, the diff is what I used to test the feature, it causes the issue in my test build (diff on boolean series). So if I remove it, the bug will not show up when testing.
Last comment, I tried using pytest and got some errors so went with the old fashioned way.

Unprocessable replied:
Apparently the warning was added in numpy 1.13.0, 3 years ago.

TomAugspurger replied:
So is the issue with `Series.between` or `Series.diff`? Let's use one or the other to keep the test focused.

Unprocessable replied:
Fixed

TomAugspurger reviewed (commented):
Needs a release note. What version of NumPy will / does this show up in?

jbrockmendel committed:
missed from #27720 (#27759)

Unprocessable committed:
Using tm to assert series differences

Adam-Klaum committed:
ENH: Validation to only allow positive integers for options (#27382)

jbrockmendel commented:
Are there other tests for diff?  If so they should be collected here or this should join them.

jbrockmendel committed:
REF: separate out invalid ops (#27735)

jbrockmendel committed:
CLN: collected cleanups from other branches (#27723)

jbrockmendel committed:
BUG: fix to_datetime(dti, utc=True) (#27733)

* BUG: fix to_datetime(dti, utc=True)

* whatsnew, suggested edits

* parametrize

* backquotes

Unprocessable committed:
Added diff bug description to changelog

Unprocessable committed:
Added issue number

Unprocessable committed:
Adding timeseries tests as well

Unprocessable committed:
Removed test_diff, it is added to test_diff.py

I copied the data over to the other file, diff tests should be in one file and the boolean test cannot be added to this one as it is not a time series.

Unprocessable committed:
Removed too much code, importing nan again

Unprocessable committed:
STYLE: formatting...

jreback commented on pandas/tests/series/test_diff.py:
```diff
@@ -0,0 +1,54 @@
+from pandas.util.testing import assert_series_equal
+
+
+class TestDiff(TestData):
```
you can drop the class TestDiff
use a fixture for ts here (it’s likely already a fixture)

can you split this up into separate tests / paramaterize

Unprocessable replied:
I would love to, but I will learn that after the merge, I do not have the time now to fix it. Rewriting the whole file will take a while, let alone that the results of the test should always be the same, which cannot be guaranteed after a rewrite.

Unprocessable committed:
Splitting into two functions

Unprocessable committed:
Split into 3 functions

Unprocessable committed:
Update test_diff.py

TomAugspurger commented on pandas/tests/series/test_diff.py:
```diff
@@ -0,0 +1,54 @@
+            result, Series(TimedeltaIndex(["NaT"] + ["1 days"] * 4), name="foo")
+        )
+        
+        # boolean series
```
Can you add a test for a mix of bool & NA? `pd.Series([True, False, np.nan, False], dtype="object')`

Unprocessable replied:
Done

Unprocessable committed:
Adding nan test

Unprocessable committed:
STYLE: Black formatting

Unprocessable committed:
Import order changed

kernc committed:
BUG: Fix windowing over read-only arrays (#27767)

jbrockmendel committed:
CLN/REF: Remove _try_cast_result, _try_coerce_and_cast_result (#27764)

charlesdong1991 committed:
BUG: pd.crosstab not working when margin and normalize are set together (#27663)

albertvillanova committed:
DOC: Validate docstring directives (#27630)

Closes gh-27629

simonjayhawkins committed:
DEPR: Removed the previously deprecated ExtensionArray._formatting_values (#27774)

bhavaniravi committed:
CI: Fix setting PATH in azure pipelines (#27787)

Unprocessable committed:
Update test_diff.py

Unprocessable committed:
Update test_diff.py

Unprocessable commented:
I cannot get the import order fixed correctly, on my PC isort does not show any changes, but the checks do show up.

jschendel committed:
DEPR: Remove previously deprecated IntervalIndex.from_intervals (#27793)

jorisvandenbossche committed:
BUG: fix construction of NonConsolidatableBlock with inconsistent ndim (#27786)

qwhelan committed:
[BLD] Add script that fails build if git tags do not exist (#27770)

mroeschke committed:
REF: pandas/core/window.py into multiple files (#27736)

charlesdong1991 committed:
TST: Add tests for groupby categorical values with axis=1 (#27788)

jbrockmendel committed:
CLN: remove nested error handling (#27792)

charlesdong1991 committed:
BUG: Add mapping for pyqt for successful package installation (#27645)

* Add mapping for pyqt

TomAugspurger commented:
For now I think it's fine to just add the new test to pandas.tests.series.test_analytics.TestSeriesAnalytics. There's a `test_npdiff` there that's related.

TomAugspurger committed:
CI: remove pytest pins (#27809)

sameshl committed:
DOC: add print statement in to_latex example (#27798)

jbrockmendel committed:
REF: separate method-pinning functions (#27811)

jbrockmendel committed:
REF: Simplify _comp_method_SERIES (#27803)

jbrockmendel committed:
CLN: Assorted Cleanups (#27791)

wesm committed:
DOC: Add expanded index descriptors for specifying for RangeIndex-as-metadata in Parquet file schema  (#25709)

crepererum committed:
PERF: break reference cycle in Index._engine (#27607)

Fixes #27585

jorisvandenbossche committed:
COMPAT: restore shape for 'invalid' Index with nd array (#27818)

jbrockmendel committed:
EA: implement+test EA.view (#27633)

lizsaret committed:
Update sample() documentation (#27833)

Clarify the definition of replace parameter

Commit:
Add example for Series.str.slice() (#27832)

* Add example for Series.str.slice()

TomAugspurger committed:
Add security policy (#27822)

TomAugspurger committed:
Add tidelift sponsor to FUNDING (#27823)

TomAugspurger commented:
@Unprocessable do you have time to update with the recommendations from https://github.com/pandas-dev/pandas/pull/27755#issuecomment-519275906?

Unprocessable committed:
Moving test_diff.py to test_analytics.py

As requested...

Unprocessable committed:
Removed test_diff.py, added to test_analytics

Unprocessable committed:
Removing tailing whitespaces

Unprocessable committed:
STYLE: Black formatting

Unprocessable committed:
Update test_analytics.py

Unprocessable committed:
Update test_analytics.py

Unprocessable committed:
Removing self.ts from code

Unprocessable committed:
Update test_analytics.py

Unprocessable committed:
Adding tm.makeTimeSeries, as in the original

Unprocessable committed:
STYLE: Black

Unprocessable commented:
Apparently there are some problems with tslibs. Apart from that it should be fine. @jbrockmendel @TomAugspurger Feel free to comment on it again or merge it. I might not get around to fixing it in the next few weeks and will definitely not re-factor a whole code base to use functions/fixtures. For a quick fix however, I'll try to make the time.

Unprocessable commented:
Just checked it, everyone seems to have that problem with pandas-master at the moment, so it is not due to my code. Please fix the master and rerun the tests if you're unsure, otherwise feel free to merge. Let's finally fix the three year old bug!

TomAugspurger committed:
CI: pytest, xdist versions (#27845)

* CI: pytest, xdist

TomAugspurger commented:
Hopefully fixed in https://github.com/pandas-dev/pandas/pull/27845.

If you can merge master & re-push it should be fixed.

On Sun, Aug 11, 2019 at 5:50 PM Unprocessable <notifications@github.com>
wrote:

> Just checked it, everyone seems to have that problem with pandas-master at
> the moment, so it is not due to my code. Please fix the master and rerun
> the tests if you're unsure, otherwise feel free to merge. Let's finally fix
> the three year old bug!
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/27755?email_source=notifications&email_token=AAKAOITLPRNNCVJZNYX77TLQECJT5A5CNFSM4IJKX662YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4BKQDQ#issuecomment-520267790>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKAOIQMFBEF6BK3MRISATTQECJT5ANCNFSM4IJKX66Q>
> .
>

Unprocessable commented:
> 
> Hopefully fixed in #27845. If you can merge master & re-push it should be fixed.
> 
> I might have some time tomorrow for some quick fixes if required. 
No, I hate to say this, but I guess you are in a different time zone and tomorrow means something different. Sorry, this will have to wait a few weeks or it will have to go as is.

TomAugspurger commented:
No rush, thanks.

On Mon, Aug 12, 2019 at 2:02 AM Unprocessable <notifications@github.com>
wrote:

> Hopefully fixed in #27845
> <https://github.com/pandas-dev/pandas/pull/27845>. If you can merge
> master & re-push it should be fixed.
>
> I might have some time tomorrow for some quick fixes if required.
> No, I hate to say this, but I guess you are in a different time zone and
> tomorrow means something different. Sorry, this will have to wait a few
> weeks or it will have to go as is.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/pandas-dev/pandas/pull/27755?email_source=notifications&email_token=AAKAOISFOX3RZMRGYZ3CXSLQEEDJ7A5CNFSM4IJKX662YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGOD4BWSIY#issuecomment-520317219>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAKAOIV5XK24JOIGOGGTKI3QEEDJ7ANCNFSM4IJKX66Q>
> .
>

TomAugspurger committed:
CI: Pin Python to 3.7.3 (#27868)

TomAugspurger committed:
BUG: Fix groupby quantile segfault (#27826)

* BUG: Fix groupby quantile segfault

Validate that q is between 0 and 1.

Closes #27470

* prettier

jbrockmendel committed:
op--> opname (#27849)

jbrockmendel committed:
CLN: short-circuit case in Block.replace (#27768)

charlesdong1991 committed:
BUG: Allow plotting boolean values (#27665)

CJStadler committed:
Avoid calling S3File.s3 (#27777)

* Avoid calling S3File.s3

When reading from s3 using fastparquet. This attribute was removed in
s3fs 0.3.0. This change avoids accessing it by using a new method
get_file_and_filesystem which returns the filesystem in addition to the
file.

jorisvandenbossche committed:
BUG: add back check for MultiIndex case and take_split_path (#27855)

* BUG: add back check for MultiIndex case and take_split_path

* update comment

* add simpler test + whatsnew

jimmycallin committed:
Added missing space to error description (#27866)

Commit:
DOC: Updated Series.items 'See also' section for clarity  (#27888)

Ankit-Dhankhar committed:
DOC: clarify see also for DataFrame.iterrows() (#27893)

sameshl committed:
DOC: Add append example in to_excel documentation (#27852)

sameshl committed:
DOC: Standardize use of "Iterate" and "Iterator" (#27871)

changed `Iterator over` --> `Iterate over`

closes #27861

jbrockmendel committed:
REF: Make CategoricalIndex comparison defer to Categorical comparison (#27769)

jbrockmendel committed:
skip test if running non-installed (#27810)

jbrockmendel committed:
CLN: match standardized dispatch logic (#27830)

jbrockmendel committed:
requested edit to comparison method (#27873)

jbrockmendel committed:
CLN: use invalid_comparison for incorrect case in Index comparison (#27879)

jbrockmendel committed:
CLN: Index.__new__ (#27883)

jbrockmendel committed:
remove unnecessary validate_for_numeric_binop (#27886)

jbrockmendel committed:
CLN: remove unnecessary dtype checks (#27889)

charlesdong1991 committed:
BUG: boxplot does not work when data has datetime column (#27846)

sparalic committed:
DOC: Add CoC to the README (#27851)

* update README file

Co-Authored-By: Marc Garcia <garcia.marc@gmail.com>

jbrockmendel committed:
BUG: fix Sparse reduction (#27890)

* BUG: fix Sparse reduction

sameshl committed:
DOC:Use of "Yields" for documentation of DataFrame.iteritems() (#27876)

TomAugspurger committed:
Revert 37 pins (#27907)

* Revert "CI: Pin Python to 3.7.3 (#27868)"

This reverts commit 2ebab98cf1f36a678256436e7a4f5149536436c9.

topper-123 committed:
PERF: Break reference cycle for all Index types (#27840)

* Generalize guards for index ref cycles

* add issue number

jbrockmendel committed:
CLN: remove never-True checks (#27908)

jbrockmendel committed:
REF: implement should_extension_dispatch (#27815)

jbrockmendel committed:
CLN: remove unnecessary validate_for_numeric_unary (#27891)

jschendel committed:
DOC: Fix section reference placement in whatsnew (#27919)

jbrockmendel committed:
TST: parametrize arithmetic tests (#27847)

jbrockmendel committed:
BUG: fix+test op(NaT, ndarray), also simplify (#27807)

jbrockmendel committed:
CLN: remove _maybe_update_attributes (#27896)

jbrockmendel committed:
CLN: simplify comparison method, docstring cleanups (#27923)

MarcoGorelli committed:
TST: Raise ValueError and suggestion to use header=None if header=-1 is pa… (#27878)

MarcoGorelli committed:
BUG: Ensure that fill_na in Categorical only replaces null values (#27932)

* Ensure that :func: in :class: only replaces null values

adamjstewart committed:
DOC: Fix typos in HDFStore docs (#27940)

charlesdong1991 committed:
VIS: Fix DataFrame.plot() produces incorrect legend markers (#27808)

TomAugspurger committed:
BUG: Merge with readonly arrays (#27946)

jbrockmendel committed:
cleanups, remove StringMixin (#27939)

jbrockmendel committed:
implement array_ops (#27936)

LiuSeeker committed:
TST: Asserts all types are caterogical in function fillna with correct error message (#27933)

* Fixes issue #13628

* Fixes issue #13628 fix

* isort Fixes #13628

* move to 1.0

jbrockmendel committed:
DOC/CLN: docstring cleanups (#27942)

TomAugspurger committed:
TST: xfail on 37, win (#27956)

* TST: xfail on 37, win

Closes https://github.com/pandas-dev/pandas/issues/27902

bhavaniravi committed:
DOC: remove savefig references from the docs v0.7.3 (#27990)

isantolin committed:
DOC: Updated table with read_fwf() (#27920)

guilherme-salome committed:
BUG: lzma is a required part of python, make it optional (#27882)

* Importing lzma when Python has been compiled without its support will
raise a warning. Substituted import lzma for helper function.

fiendish committed:
BUG: Help python csv engine read binary buffers (#27925)

* BUG: Help python csv engine read binary buffers

The file buffer given to read_csv could have been opened in
binary mode, but the python csv reader errors on binary buffers.

closes #23779

jbrockmendel committed:
CLN: missing boilerplate in Sparse op (#27910)

* CLN: missing boilerplate in Sparse op

ccharlesgb committed:
BUG: _can_use_numexpr fails when passed large Series (#27773)

* BUG: _can_use_numexpr did not handle Series case correctly

jbrockmendel committed:
TST: fix flaky xfail (#28016)

* TST: fix flaky xfail

jorisvandenbossche committed:
Fix regression in .ix fallback with IntervalIndex (#27926)

jbrockmendel committed:
BUG: TimedeltaArray - Index result.name (#27962)

* BUG: TimedeltaArray - Index result.name

* change assert to TypeErrror

jbrockmendel committed:
REF: use should_extension_dispatch for comparison method (#27912)

* REF: implement should_extension_dispatch

jbrockmendel committed:
CLN: Remove incorrect check, comment, rename (#27922)

jbrockmendel committed:
BUG: fix to_timestamp out_of_bounds (#27916)

joy-rosie committed:
BUG:  added a check for if obj is instance of type in _isna-new (#27664)

* added a check for if obj is instance of type in _isna-new

charlesdong1991 committed:
PLT: plot('line') or plot('area') produces wrong xlim in xaxis in 0.25.0 (#27993)

* Fix issue 27686

dstansby committed:
BUG: Correctly reinstate Matplotlib converters (#27481)

jbrockmendel committed:
BUG: rfloordiv with fill_value, closes#27464 (#28024)

* BUG: rfloordiv with fill_value, #27464, #26793

isantolin committed:
DOC: Corrected file description in read_fwf() (#28009)

* DOC: Corrected file description in read_fwf()

WillAyd committed:
Fix GroupBy nth Handling with Observed=False (#26419)

* Added test coverage for observed=False with ops

* Fixed issue with observed=False and nth

jorisvandenbossche committed:
DataFrame html repr: also follow min_rows setting (#27991)

* DataFrame html repr: also follow min_rows setting

* add whatsnew

sparalic committed:
DOC: Add punctuation to IntervalArray docstrings (#28043)

EunSeop committed:
DOC: Change document code prun in a row (#28029)

makbigc committed:
CI: Add pip dependence explicitly (#28008)

charlesdong1991 committed:
BUG: Correct the previous bug fixing on xlim for plotting (#28059)

* Restore change in datetime like

TomAugspurger committed:
CI: Set SHA for codecov upload (#28067)

jbrockmendel committed:
CI: disable codecov (#28065)

* CI: disable codecov

* disable

TomAugspurger committed:
DOC: Update whatsnew (#28073)

TomAugspurger committed:
TST: non-strict xfail for period test (#28072)

This was XPASSing on the MacPython builds:
https://travis-ci.org/MacPython/pandas-wheels/jobs/574706922

jbrockmendel committed:
BUG: iter with readonly values, closes #28055 (#28074)

* BUG: iter with readonly values, closes #28055

* whatsnew

TomAugspurger committed:
BUG: Fixed groupby quantile for listlike q (#27827)

* BUG: Fixed groupby quantile for listlike q

Closes #27526

sumanau7 committed:
Preserve index when setting new column on empty dataframe. (#26471)

lgmsantos committed:
BUG: Series.rename raises error on values accepted by Series construc… (#27814)

* BUG: Series.rename raises error on values accepted by Series constructor.

ianzur committed:
BUG: timedelta merge asof with tolerance (#27650)

* issue #27642 - timedelta merge asof with tolerance

mtrbean committed:
DOC: update GroupBy.head()/tail() documentation (#27844)

WuraolaOyewusi committed:
DOC: Remove alias for numpy.random.randn from the docs (#28082)

martinagvilas committed:
DOC: Fix docstrings lack of punctuation (#28031)

TomAugspurger committed:
DOC: Start 0.25.2 (#28111)

* DOC: Start 0.25.2

jbrockmendel committed:
TST: fix compression tests when run without virtualenv/condaenv (#28051)

simonjayhawkins committed:
TYPING: more type hints for io.formats.printing (#27765)

WillAyd committed:
Remove Encoding of values in char** For Labels (#27618)

steveayers124 committed:
DOC: Fix GL01 and GL02 errors in the docstrings (#27988)

jbrockmendel committed:
Change trys to checks (#28121)

bmoscon committed:
Remove outdated docstring that no longer applies (#28137)

simonjayhawkins committed:
TYPING: _pytest.mark.structures.MarkDecorator -> Callable (#28134)

WillAyd committed:
Contributing Guide for Type Hints (#27050)

realjohnward committed:
DOC: Fixes to docstrings formatting (#28096)

bhuvanakundumani committed:
DOC: Fix RangeIndex and other docstrings for missing period in summary (#28123)

jalbritt committed:
DOC: Added periods to end of docstrings in explode function (#27973)

TomAugspurger committed:
COMPAT: 3.8 compat for tests and DataFrame.query (#28101)

* COMPAT: implement visit_Constant for 3.8 compat
* Updated tests for new error messages.

simonjayhawkins committed:
TYPING: --check-untyped-defs for Index.__new__ (#28141)

drew-heenan committed:
ENH: Allow compression in NDFrame.to_csv to be a dict with optional arguments (#26023) (#26024)

katrinleinweber committed:
DOC: Harmonize column selection to bracket notation (#27562)

* Harmonize column selection to bracket notation

As suggested by https://medium.com/dunder-data/minimally-sufficient-pandas-a8e67f2a2428#46f9

simonjayhawkins committed:
TYPING: add stubs for _packer and _unpacker (#28135)

WillAyd committed:
Run clang-format on objToJSON (#28144)

jbrockmendel committed:
PERF: replace with list, closes #28084 (#28099)

TomAugspurger committed:
DOC: whatsnew for 28099 (#28154)

TomAugspurger committed:
DOC: Set 1.0.0 in index.rst (#28149)

TomAugspurger committed:
BUG: Fix groupby quantile array (#28113)

simonjayhawkins committed:
TYPING: --disallow-any-expr for HTMLFormatter.__init__ (#28140)

jbrockmendel committed:
CLN: small ops optimizations (#28036)

jbrockmendel committed:
CLN: internals.blocks cleanup, typing (#27941)

charlesdong1991 committed:
Replace with nested dict raises for overlapping keys (#27696)

simonjayhawkins committed:
TYPING: --check-untyped-defs util._decorators (#28128)

addisonlynch committed:
CLN: Use ABC classes for isinstance checks, remove unnecessary imports (#28158)

* CLN: Use ABC classes for isinstance checks, remove unnecessary imports

* Formatting repairs

simonjayhawkins committed:
TYPING: change to FrameOrSeries Alias in pandas._typing (#28173)

killerontherun1 committed:
Solving GL01,GL02 in pandas.Interval and a few mentioned in the comments (#28197)

RoeiRaz committed:
Fix slicer assignment bug (#28131)

jbrockmendel committed:
STY: whitespace before class docstringsd (#28209)

DavidRosen committed:
DOC: Example for adding a calculated column in SQL and Pandas (#28182)

* Add example for adding a calculated column

simonjayhawkins committed:
CLN: minor typos MutliIndex -> MultiIndex (#28223)

jbrockmendel committed:
reduction-> libreduction for grepability (#28184)

jschendel committed:
REGR: Fix to_csv with IntervalIndex (#28229)

* REGR: Fix to_csv with IntervalIndex

charlesdong1991 committed:
BUG: Multiple lambdas in named aggregation (#27921)

jbrockmendel committed:
PERF: lazify pytz seqToRE call, trims 35ms from import (#28228)

basnijholt committed:
STYLE: run pre-commit filters on the repo (#27915)

* add isort:skip to "from .pandas_vb_common import setup"

* add isort:skip to noqa: E402 marked lines

* run black

* add noqa: E402 isort:skip where needed

* run pre-commit filters on asv_bench/benchmarks/

* parse the isort config when using pre-commit

* run isort on pandas/core/api.py

* run pre-commit filters and commit trivial import sorting changes

* specify flake8 errors in pandas/io/msgpack/__init__.py

* fix imports for doc/source/conf.py

* fix the [isort] skip entry in setup.cfg

Also I removed the files for which I have fixed the problems.

simonjayhawkins committed:
REGR: <th> tags for notebook display closes #28204 (#28216)

* REGR: <th> tags for notebook display closes #28204

quintusdias committed:
Fix read of py27 pytables tz attribute, gh#26443 (#28221)

oguzhanogreden committed:
DOC: Document existing functionality of pandas.DataFrame.to_sql() #11886 (#26795)

* DOC: add single dtype to NDFrame.to_sql

jbrockmendel committed:
CLN: catch less inside try/except (#28203)

* CLN: catch less inside try/except

wholmgren committed:
DOC: fix DatetimeIndex.tz_localize doc string example (#28237)

Commit:
Adding updates

Commit:
BUG: Fix numpy boolean subtraction error in Series.diff (#27755)

Commit:
Merge branch 'patch-2' of https://github.com/Unprocessable/pandas into patch-2

Commit:
Resolved merge conflict added changes to older whatsnew.

Unprocessable committed:
Update v0.25.1.rst

simonjayhawkins commented:
superseded by #28251
